src/commands/art.ts

```diff
@@ -0,0 +1,127 @@
+					.setName("type")
+					.setDescription("Whether you're searching by password, Konami ID, or name.")
+					.setRequired(false)
+					.addChoice("Password", "password")
+					.addChoice("Konami ID", "kid")
+					.addChoice("Name", "name")
+			)
+			.toJSON();
+	}
+
+	protected override get logger(): Logger {
+		return this.#logger;
+	}
+
+	async getCard(type: "password" | "kid" | "name", input: string): Promise<Static<typeof CardSchema> | undefined> {
+		let url = `${process.env.SEARCH_API}`; // treated as string instead of string? without forbidden non-null check
+		input = encodeURIComponent(input);
+		if (type === "password") {
+			url += `/card/password/${input}`;
+		} else if (type === "kid") {
+			url += `/card/kid/${input}`;
+		} else {
+			url += `/art?name=${input}`;
+		}
+		const response = await fetch(url);
+		// 400: Bad syntax, 404: Not found
+		if (response.status === 400 || response.status === 404) {
+			return undefined;
+		}
+		// 200: OK
+		if (response.status === 200) {
+			return await response.json();
+		}
+		throw new Error((await response.json()).message);
+	}
+
+	async getArt(card: Static<typeof CardSchema>): Promise<FileOptions | undefined> {
+		const artUrl = `${process.env.IMAGE_HOST}/${card.password}.png`;
+		const response = await fetch(artUrl);
+		// 400: Bad syntax, 404: Not found
+		if (response.status === 400 || response.status === 404) {
+			return undefined;
+		}
+		// 200: OK
+		if (response.status === 200) {
+			const art = await response.buffer();
+			return { attachment: art };
+		}
+		throw new Error((await response.json()).message);
+	}
+
+	protected override async execute(interaction: CommandInteraction): Promise<number> {
+		let type = interaction.options.getString("type", false) as "password" | "kid" | "name" | undefined;
+		let input = interaction.options.getString("input", true);
+		if (!type) {
+			// handle edge case for specific bad input
+			if (parseInt(input).toString() === input && input !== "NaN") {
+				// if its all digits, treat as password.
+				type = "password";
+			} else if (input.startsWith("#")) {
+				// initial # indicates KID, as long as the rest is digits
+				const kid = input.slice(1);
+				if (parseInt(kid).toString() === kid && kid !== "NaN") {
+					type = "kid";
+					input = kid;
+				} else {
+					type = "name";
+				}
+			} else {
+				type = "name";
+			}
+		}
+		await interaction.deferReply();
+		const card = await this.getCard(type, input);
+		let end: number;
+		if (!card) {
+			end = Date.now();
+			// TODO: include properly-named type in this message
+			await interaction.editReply({ content: `Could not find a card matching \`${input}\`!` });
+		} else {
+			const artFile = await this.getArt(card);
+			end = Date.now();
+			if (artFile) {
+				// TODO: display name along with art?
+				await interaction.editReply({ files: [artFile] }); // Actually returns void
+			} else {
+				// TODO: do we include a lang parameter just for this niche error message?
```

Comment:
No, we'd use the locale setting previously introduced. That parameter as an override only makes sense if you implement the other to-do to display the name.